### ISSUE TITLE
Upload issue with ABP 3.3.0

### ISSUE THREAD

**Issue by realdadfish** [getsentry/sentry-java#644]
It seems as if the Android Gradle team did again break the manifest API - when I look at the `--debug` output of a failed upload task, I see that the encapsulated `.exe` file is called with

```
/var/folders/5r/5t6cr_550tqdnr6lk40pw7c4q0q8sl/T/.sentry-cli3942433204811368398.exe upload-proguard 
--android-manifest property(interface org.gradle.api.file.Directory, transform(property(interface org.gradle.api.file.Directory, fixed(class org.gradle.api.internal.file.DefaultProjectLayout$FixedDirectory, /path/to/app/build))))/AndroidManifest.xml 

--write-properties /path/to/app/build/intermediates/merged_assets/ProductionRelease/mergeProductionReleaseAssets/out/sentry-debug-meta.properties 
/path/to/app/build/outputs/mapping/Production/release/mapping.txt
```

I think your PR #608 made matters a bit worse here and this would be the proper fix for 3.0+ onwards:

```
try {
         // Android Gradle Plugin < 3.0.0
         manifestPath = variantOutput.processManifest.manifestOutputFile
} catch (Exception ignored) {
          // Android Gradle Plugin >= 3.0.0
          def outputDir = variantOutput.processManifest.manifestOutputDirectory
          // Gradle 4.7 introduced the lazy task API and AGP 3.3+ adopts that,
          // so we apparently have a Provider<File> here instead
          if (!(outputDir instanceof File)) {
              outputDir = outputDir.get().asFile
          }
          manifestPath = new File(outputDir, "AndroidManifest.xml")
}
```

This verifyable gives me back a proper `manifestPath` on AGP 3.3.0 beta04 w/ Gradle 4.10.2. You might want to have a look at [Gradle's Lazy Task](https://docs.gradle.org/current/userguide/task_configuration_avoidance.html) implementation, because the way it works now (above) is that because I query the output dir of the manifest, I implicitely also configure the manifest creation (and possibly more dependencies) for the specific build flavor.

In other words, if I have an app build with 10 flavours but only want to create a release build for one flavor, your plugin will trigger the configuration of the nine other (unused) tasks that create the merged manifests even though they're not needed for the upcoming upload, prolonging the build eventually quite a bit.

**Comment by bretthoerner** [comment]
Thanks for the report, do you want to make a PR for this?

**Comment by realdadfish** [comment]
I'd rather let you do it properly, wrt the lazy task implementation. The above three-liner is just a quick fix and I'm unable to provide anything more at the moment, sorry. Since you can override the manifest path via the plugin's configuration I also have a workaround in place for now.
